src/handler/HumidityHandler.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+*   HumidityHandler.cpp
+*   ----------------------
+*   Created on: 2023/09/26
+*   Author: Lankow
+*/
+#include "handler/HumidityHandler.hpp"
+#include "constants.hpp"
+#include "Logger.hpp"
+
+static uint8_t m_idCounter;
+
+HumidityHandler::HumidityHandler(uint8_t p_operatedPin)
+    : m_operatedPin(p_operatedPin), m_handlerId(m_idCounter++) {}
+
+void HumidityHandler::handleHumidity() {
+    auto dataProvider = getDataProvider();
+    if (!dataProvider) {
```

Comment:
jak to jest możliwe, że tu będzie nullptr skoro przekazujesz ten data provider w ctorze?